File: support_tools/swigwin-3.0.12/Examples/test-suite/go/preproc_runme.go
package main

import "./preproc"

func main() {
	if preproc.GetEndif() != 1 {
		panic(0)
	}

	if preproc.GetDefine() != 1 {
		panic(0)
	}

	if preproc.GetDefined() != 1 {
		panic(0)
	}

	if 2*preproc.One != preproc.Two {
		panic(0)
	}
}

File: support_tools/swigwin-3.0.12/Examples/test-suite/perl5/preproc_runme.pl
#!/usr/bin/perl
use strict;
use warnings;
use Test::More tests => 6;
BEGIN { use_ok('preproc') }
require_ok('preproc');

# adapted from ../python/preproc_runme.py

is($preproc::endif, 1);
is($preproc::define, 1);
is($preproc::defined, 1);
is($preproc::one * 2, $preproc::two);

